From 9163446e7c0eed6c5796c99b73ed5800ba3a3f2c Mon Sep 17 00:00:00 2001
Message-Id: <9163446e7c0eed6c5796c99b73ed5800ba3a3f2c.1652780988.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Fri, 13 May 2022 17:55:23 +0200
Subject: [PATCH] Python REPL: Shutdown CHIP stack properly

---
 src/controller/python/chip/ChipReplStartup.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/controller/python/chip/ChipReplStartup.py b/src/controller/python/chip/ChipReplStartup.py
index 861d07ca8a..efdf8ea4cb 100644
--- a/src/controller/python/chip/ChipReplStartup.py
+++ b/src/controller/python/chip/ChipReplStartup.py
@@ -12,6 +12,7 @@ import chip.logging
 import argparse
 import builtins
 import chip.FabricAdmin
+import atexit
 
 _fabricAdmins = None
 
@@ -98,6 +99,12 @@ def ReplInit(debug):
         logging.getLogger().setLevel(logging.WARN)
 
 
+def StackShutdown():
+    chip.FabricAdmin.FabricAdmin.ShutdownAll()
+    ChipDeviceCtrl.ChipDeviceController.ShutdownAll()
+    builtins.chipStack.Shutdown()
+
+
 def matterhelp(classOrObj=None):
     if (classOrObj is None):
         inspect(builtins.devCtrl, methods=True, help=True, private=False)
@@ -135,5 +142,7 @@ devCtrl = CreateDefaultDeviceController()
 
 builtins.devCtrl = devCtrl
 
+atexit.register(StackShutdown)
+
 console.print(
     '\n\n[blue]Default CHIP Device Controller has been initialized to manage [bold red]fabricAdmins[0][blue], and is available as [bold red]devCtrl')
-- 
2.36.1

