From 73d547fc5d3c1793df8955dc8db95351b59a8d26 Mon Sep 17 00:00:00 2001
Message-Id: <73d547fc5d3c1793df8955dc8db95351b59a8d26.1652462795.git.stefan@agner.ch>
From: Stefan Agner <stefan@agner.ch>
Date: Fri, 13 May 2022 17:55:23 +0200
Subject: [PATCH] Python REPL: Shutdown CHIP stack properly

---
 src/controller/python/chip/ChipReplStartup.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/src/controller/python/chip/ChipReplStartup.py b/src/controller/python/chip/ChipReplStartup.py
index 7fcb9ad294..b4008558cc 100644
--- a/src/controller/python/chip/ChipReplStartup.py
+++ b/src/controller/python/chip/ChipReplStartup.py
@@ -12,6 +12,7 @@ import chip.logging
 import argparse
 import builtins
 import chip.FabricAdmin
+import atexit
 
 _fabricAdmins = None
 
@@ -96,6 +97,12 @@ def ReplInit():
     logging.getLogger().setLevel(logging.WARN)
 
 
+def StackShutdown():
+    chip.FabricAdmin.FabricAdmin.ShutdownAll()
+    ChipDeviceCtrl.ChipDeviceController.ShutdownAll()
+    builtins.chipStack.Shutdown()
+
+
 def matterhelp(classOrObj=None):
     if (classOrObj is None):
         inspect(builtins.devCtrl, methods=True, help=True, private=False)
@@ -130,5 +137,7 @@ devCtrl = CreateDefaultDeviceController()
 
 builtins.devCtrl = devCtrl
 
+atexit.register(StackShutdown)
+
 console.print(
     '\n\n[blue]Default CHIP Device Controller has been initialized to manage [bold red]fabricAdmins[0][blue], and is available as [bold red]devCtrl')
-- 
2.36.1

